crates/revm/src/db/states/bundle_state.rs

```diff
@@ -7,6 +7,148 @@ use revm_interpreter::primitives::{
     hash_map::{self, Entry},
     AccountInfo, Bytecode, HashMap, StorageSlot, B160, B256, U256,
 };
+use std::collections::HashSet;
+
+#[derive(Debug, Default)]
+pub struct BundleBuilder {
+    states: HashSet<B160>,
+    state_original: HashMap<B160, AccountInfo>,
+    state_present: HashMap<B160, AccountInfo>,
+    state_storage: HashMap<B160, HashMap<U256, (U256, U256)>>,
+
+    reverts: HashSet<(usize, B160)>,
```

Comment:
`HashSet` would not preserve the order of reverts. So it is better to use `BTreeMap`.